libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/config/SyncsConfig.kt

```diff
@@ -0,0 +1,158 @@
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.mobilesync.config
+
+import android.content.Context
+import com.salesforce.androidsdk.mobilesync.manager.SyncManager
+import com.salesforce.androidsdk.mobilesync.target.SyncDownTarget
+import com.salesforce.androidsdk.mobilesync.target.SyncUpTarget
+import com.salesforce.androidsdk.mobilesync.util.MobileSyncLogger
+import com.salesforce.androidsdk.mobilesync.util.SyncOptions
+import com.salesforce.androidsdk.mobilesync.util.SyncState
+import com.salesforce.androidsdk.smartstore.config.StoreConfig
+import com.salesforce.androidsdk.smartstore.store.SmartStore
+import com.salesforce.androidsdk.util.JSONObjectHelper
+import com.salesforce.androidsdk.util.ResourceReaderHelper
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Class encapsulating syncs definition.
+ *
+ * Config expected in a resource or assets file in JSON with the following:
+ * {
+ * syncs: [
+ * {
+ * syncType: syncUp | syncDown
+ * syncName: xxx
+ * soupName: yyy
+ * target: { depends on target - see SyncTarget.java  }
+ * options: { also depends on target - see SyncOptions.java }
+ * }
+ * ]
+ * }
+ */
+class SyncsConfig private constructor(str: String?) {
+    private var syncConfigs: JSONArray? = null
+
+    /**
+     * Constructor for config stored in resource file
+     * @param ctx Context.
+     * @param resourceId Id of resource file.
+     */
+    constructor(ctx: Context?, resourceId: Int) : this(
+        ResourceReaderHelper.readResourceFile(
+            ctx,
+            resourceId
+        )
+    )
+
+    /**
+     * Constructor for config stored in asset file
+     * @param ctx Context.
+     * @param assetPath Path of assets file.
+     */
+    constructor(ctx: Context?, assetPath: String?) : this(
+        ResourceReaderHelper.readAssetFile(
+            ctx,
+            assetPath
+        )
+    )
+
+    init {
+        try {
+            syncConfigs = if (str == null) {
+                null
+            } else {
+                val config = JSONObject(str)
+                config.getJSONArray(SYNCS)
+            }
+        } catch (e: JSONException) {
+            MobileSyncLogger.e(TAG, "Unhandled exception parsing json", e)
+        }
+    }
+
+    /**
+     * Return true if syncs are defined in config
+     * @return
+     */
+    fun hasSyncs(): Boolean {
+        return syncConfigs?.let { it.length() > 0 } == true
+    }
+
+    /**
+     * Create the syncs from the config in the given store
+     * NB: only feedback is through the logs - the config is static so getting it right is something the developer should do while writing the app
+     * @param store
+     */
+    fun createSyncs(store: SmartStore) {
+        val syncConfigs = syncConfigs ?: return
+        val syncManager: SyncManager = SyncManager.getInstance(null, null, store)
```

Comment:
Optional: After the Kotlin conversion, a lot of explicit types can be removed, such as `SyncManager` here.